server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -864,6 +873,29 @@ private SchemaInfo normalizeSchemaBinary(SchemaInfo schemaInfo) {
         return new SchemaInfo(schemaInfo.getType(), schemaInfo.getSerializationFormat(), schemaBinary, schemaInfo.getProperties());
     }
 
+    private void validateJsonSchema(String schemaString) {
+        try {
+            // 1. try draft 3
+            // jackson JsonSchema only supports json draft 3. If the schema definition is not compatible with draft 3, 
+            // try parsing the schema with everit library which supports drafts 4 6 and 7. 
+            OBJECT_MAPPER.readValue(schemaString, JsonSchema.class);
+        } catch (IOException e) {
+            handleDraft4onward(schemaString);
+        }
+    }
+
+    private void handleDraft4onward(String schemaString) {
```

Comment:
This method could be named better to reflect its intent. Maybe something like `validateJsonSchema4Onwards()`